neqo-transport/src/crypto.rs

```diff
@@ -1358,6 +1360,80 @@ pub enum CryptoStreams {
     },
 }
 
+/// [Fisher–Yates shuffle](https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle#Modern_method)
+/// Modified to make sure no element stays in place if `a` has more than one element.
+///
+/// TODO: We could use <https://docs.rs/rand/latest/rand/seq/trait.SliceRandom.html#tymethod.shuffle>
+/// instead, but I assume we want to only use the NSS RNG in neqo?
+fn shuffle<T>(a: &mut [T]) {
+    // To shuffle an array a of n elements (indices 0..n-1)
+    let n: usize = a.len();
+    if n < 2 {
+        return;
+    }
+    for i in 0..(n - 1) {
+        // j ← random integer such that i ≤ j < n
+        let j = loop {
+            let j = (random::<8>()[0] as usize) % (n - i) + i;
+            if j != i {
+                break j;
+            }
+        };
+        // Exchange a[i] and a[j]
+        a.swap(i, j);
+    }
+}
+
+/// Find the start and end indices of all sequences of `len` or more graphic ASCII bytes in `data`.
+fn ascii_sequences(data: &[u8], len: usize) -> Vec<(usize, usize)> {
```

Comment:
```suggestion
fn ascii_sequences(data: &[u8], len: usize) -> Vec<Range<usize>> {
```

Then you can use `start..end` for each value.